catkit2/services/thorlabs_cube_motor/thorlabs_cube_motor.py

```diff
@@ -23,36 +22,31 @@ def __init__(self):
 
     def open(self):
         """
-            Open the motor and cube.
+        Open connexion to the motor.
 
-            This will also check if the configured characteristics are matching the one the cube have.
+        This will also check if the configured characteristics are matching the one the cube have.
         """
         self.motor = apt.Motor(self.cube_serial_number)
 
-        # Retrieve min, max, unit and model of the motor and cube
+        # Retrieve the min, max, unit and model from the connected device.
         min = self.motor.get_stage_axis_info()[0]
         max = self.motor.get_stage_axis_info()[1]
-        unit = ''
+        unit = None
         model = self.motor.hardware_info[0][2:-1]
         if self.motor.get_stage_axis_info()[2] == 1:
             unit = 'mm'
         if self.motor.get_stage_axis_info()[2] == 2:
             unit = 'deg'
 
-        # Config min, max, unit and model of the motor and cube
+        # Read min, max, unit and model from service configuration.
         self.min_pos = self.config['motor_min_pos']
         self.max_pos = self.config['motor_max_pos']
         self.unit = self.config['motor_unit']
         self.cube_model = self.config['cube_model']
 
-        # Compare
-        try:
-            self.min_pos == min
-            self.max_pos == max
-            self.unit == unit
-            self.cube_model == model
-        except Exception:
-            raise
+        # Compare the device parameters to the service configuration.
+        if not (self.min_pos == min and self.max_pos == max and self.unit == unit and self.cube_model == model):
+            ValueError("Device parameters don't match configuration parameters.")

```

Comment:
An exception needs to be raised, you will need to change it to look like this (I had given you a wrong example in my original review comment, sorry about that):
```python
raise ValueError("Device parameters don't match configuration parameters.")
```